packages/components/src/components/post-tooltip/post-tooltip.tsx

```diff
@@ -0,0 +1,231 @@
+    this.tooltipRef.addEventListener('beforetoggle', this.handleToggle.bind(this));
+
+    // Initially position tooltip to prevent a flash of unpositioned tooltip
+    this.positionTooltip();
+  }
+
+  /**
+   * Show this tooltip
+   */
+  @Method()
+  async showTooltip() {
+    this.tooltipRef.showPopover();
+  }
+
+  /**
+   * Hide this tooltip
+   */
+  @Method()
+  async hideTooltip() {
+    this.tooltipRef.hidePopover();
+  }
+
+  /**
+   * Toggle tooltip display
+   * @param force Pass true to always show or false to always hide
+   */
+  @Method()
+  async toggleTooltip(force?: boolean) {
+    this.tooltipRef.togglePopover(force);
+  }
+
+  /**
+   * Start or stop auto updates based on tooltip events.
+   * Tooltips can be closed or opened with other methods than class members,
+   * therefore listening to the toggle event is safer for cleaning up.
+   * @param e ToggleEvent
+   */
+  private handleToggle(e: ToggleEvent) {
+    const isOpen = e.newState === 'open';
+    if (isOpen) {
+      this.startAutoupdates();
+    } else {
+      if (typeof this.clearAutoupdate === 'function') this.clearAutoupdate();
+    }
+  }
+
+  /**
+   * Start listening for DOM updates, scroll events etc. that have
+   * an influence on tooltip positioning
+   */
+  private startAutoupdates() {
+    this.clearAutoupdate = autoUpdate(
+      this.trigger,
+      this.tooltipRef,
+      this.positionTooltip.bind(this),
+    );
+  }
+
+  // Tooltip and arrow positioning with floating-ui
+  // Docs: https://floating-ui.com/docs/computePosition
+  private async positionTooltip() {
+    const {
+      x,
+      y,
+      middlewareData,
+      placement: currentPlacement,
+    } = await computePosition(this.trigger, this.tooltipRef, {
+      placement: this.placement || 'top',
+      middleware: [
+        flip(),
+        inline(),
+        shift({ padding: 8 }),
+        offset(12), // 4px outside of element to account for focus outline + ~arrow size
+        arrow({ element: this.arrowRef, padding: 8 }),
+      ],
+    });
+
+    // Tooltip
+    this.tooltipRef.style.left = `${x}px`;
+    this.tooltipRef.style.top = `${y}px`;
+
+    // Arrow
+    // Tutorial: https://codesandbox.io/s/mystifying-kare-ee3hmh?file=/src/index.js
+    const side = currentPlacement.split('-')[0];
+    const { x: arrowX, y: arrowY } = middlewareData.arrow;
+    const staticSide = sidemap[side];
+
+    Object.assign(this.arrowRef.style, {
+      top: arrowY ? `${arrowY}px` : '',
+      left: arrowX ? `${arrowX}px` : '',
+      [staticSide]: `${-this.arrowRef.offsetWidth / 2}px`,
+    });
+  }
+
+  render() {
+    return (
+      <Host data-version={version}>
+        <p
+          role="tooltip"
+          ref={(el: HTMLParagraphElement & IPopoverElement) => (this.tooltipRef = el)}
+        >
+          <span
+            class="arrow"
+            ref={el => {
+              this.arrowRef = el;
+            }}
+          ></span>
+          <slot>Hi there 👋</slot>
+        </p>
```

Comment:
Why we're not simply using a div as the tooltip, as we anyway change the role?
If at all, I would wrap only the slot with a paragraph tag:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role#examples

Reply:
Using a div now (using a dialog is really not working due to the specific semantics of dialog)